src/Evolution/DgSubcell/Matrices.cpp

```diff
@@ -20,11 +20,76 @@
 #include "Utilities/GenerateInstantiations.hpp"
 #include "Utilities/Gsl.hpp"
 #include "Utilities/Numeric.hpp"  // IWYU pragma: keep
+#include "Utilities/StaticCache.hpp"
 
 namespace evolution::dg::subcell::fd {
+template <size_t Dim>
+const Matrix& projection_matrix(const Mesh<Dim>& dg_mesh,
+                                const Index<Dim>& subcell_extents) {
+  ASSERT(dg_mesh.basis(0) == Spectral::Basis::Legendre,
+         "FD Subcell projection only supports Legendre basis right now but got "
+         "basis "
+             << dg_mesh.basis(0));
+  ASSERT(dg_mesh == Mesh<Dim>(dg_mesh.extents(0), dg_mesh.basis(0),
+                              dg_mesh.quadrature(0)),
+         "The mesh must be uniform but is " << dg_mesh);
+  ASSERT(subcell_extents == Index<Dim>(subcell_extents[0]),
+         "The subcell mesh must be uniform but is " << subcell_extents);
+  switch (dg_mesh.quadrature(0)) {
+    case Spectral::Quadrature::GaussLobatto: {
+      static const auto cache_gl = make_static_cache<
+          CacheRange<
+              Spectral::minimum_number_of_points<
+                  Spectral::Basis::Legendre,
+                  Spectral::Quadrature::GaussLobatto>,
+              Spectral::maximum_number_of_points<Spectral::Basis::Legendre>>,
```

Comment:
The `End` of a `CacheRange` is exclusive. Do you want to add a `+1` here (and below)? If no, please add a comment.